src/main/java/io/gatling/mojo/AbstractEnterprisePluginMojo.java

```diff
@@ -0,0 +1,119 @@
+
+import io.gatling.plugin.EnterprisePlugin;
+import io.gatling.plugin.EnterprisePluginClient;
+import io.gatling.plugin.InteractiveEnterprisePlugin;
+import io.gatling.plugin.InteractiveEnterprisePluginClient;
+import io.gatling.plugin.client.EnterpriseClient;
+import io.gatling.plugin.client.http.OkHttpEnterpriseClient;
+import io.gatling.plugin.exceptions.UnsupportedClientException;
+import io.gatling.plugin.io.JavaPluginScanner;
+import io.gatling.plugin.io.PluginIO;
+import io.gatling.plugin.io.PluginLogger;
+import io.gatling.plugin.io.PluginScanner;
+import java.net.URL;
+import java.util.Scanner;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugin.logging.Log;
+import org.apache.maven.plugins.annotations.Parameter;
+
+public abstract class AbstractEnterprisePluginMojo extends AbstractEnterpriseMojo {
+
+  @Parameter(defaultValue = "https://cloud.gatling.io")
+  protected URL enterpriseUrl;
+
+  @Parameter(
+      defaultValue = "${env.GATLING_ENTERPRISE_API_TOKEN}",
+      property = "gatling.enterprise.apiToken")
+  protected String apiToken;
+
+  private final Log logger = getLog();
+  private final PluginLogger pluginLogger =
+      new PluginLogger() {
+        @Override
+        public void info(String message) {
+          logger.info(message);
+        }
+
+        @Override
+        public void error(String message) {
+          logger.error(message);
+        }
+      };
+
+  private final Scanner scanner = new Scanner(System.in);
+  private final PluginScanner pluginScanner = new JavaPluginScanner(scanner);
+
+  private final PluginIO pluginIO =
+      new PluginIO() {
+        @Override
+        public PluginLogger getLogger() {
+          return pluginLogger;
+        }
+
+        @Override
+        public PluginScanner getScanner() {
+          return pluginScanner;
+        }
+      };
+
+  protected EnterprisePlugin initEnterprisePlugin() throws MojoFailureException {
+    EnterpriseClient enterpriseClient = initEnterpriseClient();
+    return new EnterprisePluginClient(enterpriseClient);
+  }
+
+  protected InteractiveEnterprisePlugin initInteractiveEnterprisePlugin()
+      throws MojoFailureException {
+    EnterpriseClient enterpriseClient = initEnterpriseClient();
+    return new InteractiveEnterprisePluginClient(enterpriseClient, pluginIO);
+  }
+
+  private EnterpriseClient initEnterpriseClient() throws MojoFailureException {
+    if (apiToken == null) {
+      final String msg =
+          "Missing API token\n"
+              + "An API token is required to call the Gatling Enterprise server; see https://gatling.io/docs/enterprise/cloud/reference/admin/api_tokens/ and create a token with the role 'Configure'.\n"
+              + CommonLogMessage.missingConfiguration(
+                  "API token",
+                  "apiToken",
+                  "gatling.enterprise.apiToken",
+                  "GATLING_ENTERPRISE_API_TOKEN",
+                  "MY_API_TOKEN_VALUE");
```

Comment:
For error messages like this one, the first line (`Missing API token`) isn't very useful when there's a more complete explanation on the second line. However, due to the way Maven displays errors, the beginning of the error message isn't always easy to find/read in the console... So I mostly just wanted to add a line break before any useful information.